apps/wpcom-block-editor/src/common/rich-text.js

```diff
@@ -4,37 +4,54 @@
  * External dependencies
  */
 import { compose, ifCondition } from '@wordpress/compose';
-import { withSelect, withDispatch } from '@wordpress/data';
+import { withSelect, withDispatch, select as baseSelect } from '@wordpress/data';
 import { RichTextToolbarButton } from '@wordpress/editor';
-import { toggleFormat, registerFormatType } from '@wordpress/rich-text';
-import { get } from 'lodash';
+import { toggleFormat, registerFormatType, unregisterFormatType } from '@wordpress/rich-text';
+import { get, find } from 'lodash';
 
-registerFormatType( 'wpcom/underline', {
-	title: wpcomGutenberg.richTextToolbar.underline,
-	tagName: 'span',
-	className: null,
-	attributes: { style: 'style' },
-	edit( { isActive, value, onChange } ) {
-		const onToggle = () =>
-			onChange(
-				toggleFormat( value, {
-					type: 'wpcom/underline',
-					attributes: {
-						style: 'text-decoration: underline;',
-					},
-				} )
-			);
+// trying to wait until core registers their keyboard-only underline,
+// then unregistering it and registering our own so they don't collide
+let times = 0;
+const maxTimes = 10;
+const { getFormatTypes } = baseSelect( 'core/rich-text' );
+const interval = setInterval( () => {
+	times = times + 1;
+	if ( times > maxTimes ) {
+		clearInterval( interval );
+		return;
+	}
+	if ( ! find( getFormatTypes(), { name: 'core/underline' } ) ) {
```

Comment:
@mmtr or @dmsnell may have some insight into this, we take over the core/video registration for VideoPress

Reply:
Hmm, I guess that was https://github.com/Automattic/jetpack/pull/11193 but not sure if this is similar

Reply:
I would assume that `core/video` refers to an entire block, rather than a `formatType` like this?

Reply:
It's an entire block so probably not applicable here. Unless @ellatrix notes an alternative, this is okay for a short term solution, but we should follow up with a better hook in Gutenberg.

Also it might be a bit safer to use `setTimeout` and not queue the next one if we're past 10 tries.

Reply:
We use the [`blocks.registerBlockType`](https://github.com/WordPress/gutenberg/blob/b612fe8616ce774af2ff1084af2e9af8b3de1de2/packages/blocks/src/api/registration.js#L101) filter to enhance the core video block with VideoPress support:

https://github.com/Automattic/jetpack/blob/9b0d1bb2ec67793e35ab0680db9cdb85d9b73608/extensions/blocks/videopress/editor.js#L119

Unfortunately, Gutenberg doesn't provide a similar filter for the format types allowing us to extend them. I took the opportunity to open a PR in Gutenberg proposing that hook:

https://github.com/WordPress/gutenberg/pull/15428

Not a blocker for this PR though. We can follow up later if the `richText.registerFormatType` hook gets merged to Gutenberg.

Reply:
In the meantime, instead of using `setInterval` we can use the subscriptions to the state to detect when the `core/underline` format is registered: 

```js
const unsubscribe = subscribe( () => {
	const underlineFormat = select( 'core/rich-text' ).getFormatType( 'core/underline' );
	if ( ! underlineFormat ) {
		return;
	}
	unsubscribe();
	unregisterFormatType( 'core/underline' );
	registerFormatType( 'wpcom/underline', ... );
} );
```

Reply:
Extra props to the `subscribe` solution because it eliminates a data race instead of making it slightly-less-likely